### PR TITLE
HTBHF-2682 Removed the ability to update a claim.

### PR DESCRIPTION
**Pull request by dwybourn**
Removed the ability to update a claim. Now any claim that has a nino matching an existing active claim will return a duplicate response and be rejected.
Due to there only being one ClaimService (no v2/v3), the lose of update functionality is affecting both the v2 and v3 endpoints.

### CODE REVIEW

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java [hunk 1]
```diff
@@ -53,38 +49,29 @@
     public ClaimResult createOrUpdateClaim(ClaimRequest claimRequest) {
```
Comment (YetAnotherMatt) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java]: We should rename this method, and the similarly named method in the claim controller

Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java]: Added TODO to rename in it's own PR.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java [hunk 2]
```diff
@@ -50,41 +46,33 @@
...
-            if (decision.claimExistsAndIsEligible()) {
-                Claim claim = claimRepository.findClaim(decision.getExistingClaimId());
-                List<UpdatableClaimantField> updatedFields = updateClaim(claim, claimRequest.getClaimant());
-                sendAdditionalPaymentMessageIfNewDueDateProvided(claim, updatedFields);
-                VoucherEntitlement weeklyEntitlement = decision.getVoucherEntitlement().getFirstVoucherEntitlementForCycle();
-                claimMessageSender.sendReportClaimMessageWithUpdatedClaimantFields(claim, decision.getDateOfBirthOfChildren(), updatedFields);
-                return ClaimResult.withEntitlementAndUpdatedFields(claim, weeklyEntitlement, updatedFields, decision.getIdentityAndEligibilityResponse());
+            if (decision.getEligibilityStatus() == EligibilityStatus.DUPLICATE) {
+                Claim claim = createAndSaveClaim(claimRequest, decision);
+                claimMessageSender.sendReportClaimMessage(claim, emptyList(), REJECTED);
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java]: There are a number of different enums being used in this class, can we not use any enums statically in this class as I have no idea which enum `REJECTED` is referring to without opening the code up in IntelliJ.

Comment (mikeSmith08) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java]: Also, should this be specifically `DUPLICATE` rather than `REJECTED`?

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java [hunk 3]
```diff
@@ -50,41 +46,33 @@
...
+                return ClaimResult.withNoEntitlement(claim);
             }
 
             Claim claim = createAndSaveClaim(claimRequest, decision);
             if (claim.getClaimStatus() == ClaimStatus.NEW) {
                 claimMessageSender.sendInstantSuccessEmailMessage(claim, decision);
                 claimMessageSender.sendNewCardMessage(claim, decision);
                 VoucherEntitlement weeklyEntitlement = decision.getVoucherEntitlement().getFirstVoucherEntitlementForCycle();
-                claimMessageSender.sendReportClaimMessage(claim, decision.getDateOfBirthOfChildren(), ClaimAction.NEW);
+                claimMessageSender.sendReportClaimMessage(claim, decision.getDateOfBirthOfChildren(), NEW);
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java]: Again here, please revert this as its much clearer without the static import of the enum values.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java [hunk 4]
```diff
@@ -50,41 +46,33 @@
...
                 return ClaimResult.withEntitlement(claim, weeklyEntitlement, decision.getIdentityAndEligibilityResponse());
             }
 
-            claimMessageSender.sendReportClaimMessage(claim, decision.getDateOfBirthOfChildren(), ClaimAction.REJECTED);
+            claimMessageSender.sendReportClaimMessage(claim, decision.getDateOfBirthOfChildren(), REJECTED);
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java]: Ditto.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/v3/EligibilityAndEntitlementServiceV3.java [hunk 5]
```diff
@@ -40,15 +42,15 @@
     public EligibilityAndEntitlementDecision evaluateNewClaimant(Claimant claimant) {
         log.debug("Looking for live claims for the given NINO");
         Optional<UUID> liveClaimsWithNino = claimRepository.findLiveClaimWithNino(claimant.getNino());
+        if (liveClaimsWithNino.isPresent()) {
+            return buildDuplicateDecisionWithExistingClaimId(liveClaimsWithNino.get());
+        }
```
Comment (mikeSmith08) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/v3/EligibilityAndEntitlementServiceV3.java]: I don't understand why this has changed because of removing the `updateClaim` functionality?

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceV3IntegrationTests.java [hunk 6]
```diff
@@ -12,29 +12,42 @@
...
+import static uk.gov.dhsc.htbhf.claimant.testsupport.ClaimDTOV3TestDataFactory.aValidClaimDTOWithNoNullFields;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.ClaimTestDataFactory.aValidClaimBuilder;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.ClaimantDTOV3TestDataFactory.aClaimantDTOWithNino;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.ClaimantTestDataFactory.aClaimantWithNino;
+import static uk.gov.dhsc.htbhf.claimant.testsupport.ClaimantTestDataFactory.aValidClaimantInSameHouseholdBuilder;
 import static uk.gov.dhsc.htbhf.claimant.testsupport.VerificationResultTestDataFactory.anAllMatchedVerificationResult;
 import static uk.gov.dhsc.htbhf.claimant.testsupport.VoucherEntitlementDTOTestDataFactory.aValidVoucherEntitlementDTO;
+import static uk.gov.dhsc.htbhf.eligibility.model.EligibilityStatus.DUPLICATE;
 import static uk.gov.dhsc.htbhf.eligibility.model.EligibilityStatus.ELIGIBLE;
+import static uk.gov.dhsc.htbhf.eligibility.model.testhelper.CombinedIdAndEligibilityResponseTestDataFactory.anIdMatchedEligibilityConfirmedUCResponseWithAllMatches;
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceV3IntegrationTests.java]: How did you get this past Checkstyle, its 167 characters long? In fact, how is the build passing with it this long too?

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceV3IntegrationTests.java [hunk 7]
```diff
@@ -77,6 +89,78 @@ void shouldAcceptAndCreateANewValidClaimWithNoNullFields() throws JsonProcessing
         verifyPostToEligibilityService();
     }
 
+    @Test
+    void shouldReturnDuplicateStatusWhenEligibleClaimAlreadyExistsForDwpHouseholdIdentifier() throws JsonProcessingException {
+        //Given
+        String householdIdentifier = "dwpHousehold1";
+        ClaimDTO dto = aValidClaimDTO();
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceV3IntegrationTests.java]: This be  a `ClaimDTOV3`.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceV3IntegrationTests.java [hunk 8]
```diff
@@ -77,6 +89,78 @@ void shouldAcceptAndCreateANewValidClaimWithNoNullFields() throws JsonProcessing
         verifyPostToEligibilityService();
     }
 
+    @Test
+    void shouldReturnDuplicateStatusWhenEligibleClaimAlreadyExistsForDwpHouseholdIdentifier() throws JsonProcessingException {
+        //Given
+        String householdIdentifier = "dwpHousehold1";
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceV3IntegrationTests.java]: You can use the `DWP_HOUSEHOLD_IDENTIFIER` test constant, it is exactly this value.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceV3IntegrationTests.java [hunk 9]
```diff
@@ -77,6 +89,78 @@ void shouldAcceptAndCreateANewValidClaimWithNoNullFields() throws JsonProcessing
...
+        //Then
+        assertDuplicateResponse(response);
+        verifyPostToEligibilityService();
+    }
+
+    @Test
+    void shouldReturnDuplicateStatusWhenEligibleClaimAlreadyExistsForHmrcHouseholdIdentifier() throws JsonProcessingException {
+        //Given
+        String householdIdentifier = "hmrcHousehold1";
+        ClaimDTO dto = aValidClaimDTO();
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceV3IntegrationTests.java]: Also `ClaimDTOV3`.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceV3IntegrationTests.java [hunk 10]
```diff
@@ -77,6 +89,78 @@ void shouldAcceptAndCreateANewValidClaimWithNoNullFields() throws JsonProcessing
...
+
+        //Then
+        assertDuplicateResponse(response);
+        verifyPostToEligibilityService();
+    }
+
+    @Test
+    void shouldReturnDuplicateStatusWhenEligibleClaimAlreadyExistsForHmrcHouseholdIdentifier() throws JsonProcessingException {
+        //Given
+        String householdIdentifier = "hmrcHousehold1";
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceV3IntegrationTests.java]: You can use the `HMRC_HOUSEHOLD_IDENTIFIER` test constant, it is exactly this value.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/v3/EligibilityAndEntitlementServiceV3Test.java [hunk 11]
```diff
@@ -68,35 +65,18 @@
...
         EligibilityAndEntitlementDecision decision = eligibilityAndEntitlementServiceV3.evaluateNewClaimant(CLAIMANT);
 
         //Then
-        assertThat(decision).isEqualTo(decisionResponse);
-        verifyCommonMocks(existingClaimId, NOT_DUPLICATE);
-        verifyNoInteractions(duplicateClaimChecker);
-    }
-
-    @Test
-    void shouldReturnEligibleWhenNotDuplicateAndEligible() {
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/v3/EligibilityAndEntitlementServiceV3Test.java]: I don't understand why this test has been removed, I think this stems back to why the changes have been made to this service, it feels like if you wanted to change this it should have been in its own PR, it doesn't seem to have anything to do with remove update functionality.

Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/v3/EligibilityAndEntitlementServiceV3Test.java]: It hasn't been removed, the diff on github is showing it that way. If you scroll down you can see the test is still there.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/v3/EligibilityAndEntitlementServiceV3.java (hunk 5 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/v3/EligibilityAndEntitlementServiceV3.java]: In the case of a duplicate nino there's no need to perform the eligibility check as we already know it's a duplicate. So the live nino check has been moved up in the method.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java (hunk 2 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java]: Updated the enums to be fully qualified. REJECTED is the claim status which doesn't have a DUPLICATE status.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java (hunk 3 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java]: Done.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java (hunk 4 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java]: Done.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceV3IntegrationTests.java (hunk 8 above)
Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceV3IntegrationTests.java]: Done.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceV3IntegrationTests.java (hunk 7 above)
Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceV3IntegrationTests.java]: Updated to ClaimDTOV3

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceV3IntegrationTests.java (hunk 10 above)
Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceV3IntegrationTests.java]: Done.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceV3IntegrationTests.java (hunk 9 above)
Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceV3IntegrationTests.java]: Done.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceV3IntegrationTests.java (hunk 6 above)
Comment (YetAnotherMatt) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceV3IntegrationTests.java]: I think checkstyle skips imports when checking line length?

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/v3/EligibilityAndEntitlementServiceV3.java (hunk 5 above)
Comment (mikeSmith08) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/v3/EligibilityAndEntitlementServiceV3.java]: I think my point was that this should probably have been in its own PR, it sticks out like a sore thumb in this PR.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceV3IntegrationTests.java (hunk 6 above)
Comment (mikeSmith08) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/ClaimantServiceV3IntegrationTests.java]: Apparently you are right, not sure how I didn't spot that, I did a fair bit of faffing around because I thought this would fail.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java (hunk 2 above)
Comment (mikeSmith08) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/claim/ClaimService.java]: Ok thanks for clarifying.
